File: liband/src/main/java/com/liband/room/converters/TimestampConverter.java
package com.liband.room.converters;

import androidx.room.TypeConverter;


import java.text.DateFormat;
import java.text.ParseException;
import java.util.Date;


public class TimestampConverter {

    @TypeConverter
    public static Date fromTimestamp(DateFormat dateFormat, String value) {
        if (value != null) {
            try {
                return dateFormat.parse(value);
            } catch (ParseException e) {
                e.printStackTrace();
            }
            return null;
        } else {
            return null;
        }
    }

    @TypeConverter
    public static String dateToTimestamp(DateFormat dateFormat, Date value) {

        return value == null ? null : dateFormat.format(value);
    }
}

File: liband/src/main/java/com/liband/room/converters/TimestampConverter.kt
package com.liband.room.converters


import androidx.room.TypeConverter
import java.text.DateFormat
import java.text.ParseException
import java.util.*


object TimestampConverter {

    @TypeConverter
    fun fromTimestamp(dateFormat: DateFormat, value: String?): Date? {
        if (value != null) {
            try {
                return dateFormat.parse(value)
            } catch (e: ParseException) {
                e.printStackTrace()
            }

            return null
        } else {
            return null
        }
    }

    @TypeConverter
    fun dateToTimestamp(dateFormat: DateFormat, value: Date?): String? {

        return if (value == null) null else dateFormat.format(value)
    }
}
